### PR TITLE
Add partial template support

### LINKED ISSUE #73: [Feature request] Allow new partial templates to be registered
I naïvely added a new template in the partials folder of my project and tried to called it from the Class template.
I receive the following error at runtime:
>Referenced partial name _partial_name_ could not be resolved

### PR DESCRIPTION
**Pull request by gejotas**
This contains the code by Adrian, Mehmke with some improvement to reduce code duplication and scanning of only *.hbs.

This should resolve #73

### CODE REVIEW

File: src/EntityFrameworkCore.Scaffolding.Handlebars/EntityFrameworkCore.Scaffolding.Handlebars.csproj [hunk 1]
```diff
@@ -73,6 +73,9 @@
     <None Update="CodeTemplates\CSharpDbContext\Partials\DbConstructor.hbs">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
+    <None Update="CodeTemplates\CSharpDbContext\Partials\DbOnConfiguring.hbs">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
```
Comment (tonysneed) [inline src/EntityFrameworkCore.Scaffolding.Handlebars/EntityFrameworkCore.Scaffolding.Handlebars.csproj:78]: You can remove this file.

File: src/EntityFrameworkCore.Scaffolding.Handlebars/CodeTemplates/CSharpDbContext/DbContext.hbs [hunk 2]
```diff
@@ -14,7 +14,8 @@ namespace {{namespace}}
 
 {{{> dbconstructor}}}
 
-{{{on-configuring}}}
+{{{> dbonconfiguring}}}
+
```
Comment (tonysneed) [inline src/EntityFrameworkCore.Scaffolding.Handlebars/CodeTemplates/CSharpDbContext/DbContext.hbs:18]: Please revert changes to the **CSharpDbContext/DbContext.hbs** file.

File: src/EntityFrameworkCore.Scaffolding.Handlebars/CodeTemplates/CSharpDbContext/Partials/DbOnConfiguring.hbs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿{{spaces 8}}protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
```
Comment (tonysneed) [inline src/EntityFrameworkCore.Scaffolding.Handlebars/CodeTemplates/CSharpDbContext/Partials/DbOnConfiguring.hbs:1]: Remove file.

File: src/EntityFrameworkCore.Scaffolding.Handlebars/CodeTemplates/TypeScriptDbContext/DbContext.hbs [hunk 4]
```diff
@@ -14,7 +14,8 @@ namespace {{namespace}}
 
 {{{> dbconstructor}}}
 
-{{{on-configuring}}}
+{{{> dbonconfiguring}}}
+
```
Comment (tonysneed) [inline src/EntityFrameworkCore.Scaffolding.Handlebars/CodeTemplates/TypeScriptDbContext/DbContext.hbs:18]: Please revert changes to the **TypeScriptDbContext/DbContext.hbs** file.

File: src/EntityFrameworkCore.Scaffolding.Handlebars/CodeTemplates/TypeScriptDbContext/Partials/DbOnConfiguring.hbs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿{{spaces 8}}protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
```
Comment (tonysneed) [inline src/EntityFrameworkCore.Scaffolding.Handlebars/CodeTemplates/TypeScriptDbContext/Partials/DbOnConfiguring.hbs:1]: Please remove this file.

File: src/EntityFrameworkCore.Scaffolding.Handlebars/InMemoryTemplateFileService.cs [hunk 6]
```diff
@@ -35,6 +37,34 @@ public virtual string[] InputFiles(params InputFile[] files)
             return filePaths.ToArray();
         }
 
+        public string[] RetrieveAllFileNames(string relativeDirectory)
```
Comment (tonysneed) [inline src/EntityFrameworkCore.Scaffolding.Handlebars/InMemoryTemplateFileService.cs:40]: Missing XML comment. Copy from `ITemplateFileService`.

File: src/EntityFrameworkCore.Scaffolding.Handlebars/HbsCSharpDbContextGenerator.cs [hunk 7]
```diff
@@ -162,55 +162,20 @@ protected override void GenerateOnConfiguring(string connectionString, bool supp
         {
             Check.NotNull(connectionString, nameof(connectionString));
 
-            var sb = new IndentedStringBuilder();
```
Comment (tonysneed) [inline src/EntityFrameworkCore.Scaffolding.Handlebars/HbsCSharpDbContextGenerator.cs:165]: I like the idea of templating the `DbContext.OnConfiguring` method, but this belongs in another PR. The purpose of this PR should be simply to enable arbitrary partial templates to be registered.

**Comment by tonysneed** [comment]
When I run the generator I get an error:
```
HandlebarsDotNet.HandlebarsRuntimeException: Referenced partial name dbimports could not be resolved   at HandlebarsDotNet.Compiler.PartialBinder.InvokePartialWithFallback(String partialName, BindingContext context, HandlebarsConfiguration configuration)
   at lambda_method(Closure , TextWriter , Object )
   at HandlebarsDotNet.Handlebars.HandlebarsEnvironment.<>c__DisplayClass7_0.<Compile>b__0(Object context)
   at EntityFrameworkCore.Scaffolding.Handlebars.HbsDbContextTemplateService.GenerateDbContext(Object data) in C:\Users\Tony\Source\EntityFrameworkCore.Scaffolding.Handlebars\src\EntityFrameworkCore.Scaffolding.Handlebars\HbsDbContextTemplateService.cs:line 44
   at EntityFrameworkCore.Scaffolding.Handlebars.HbsCSharpDbContextGenerator.WriteCode(IModel model, String contextName, String connectionString, String contextNamespace, String modelNamespace, Boolean useDataAnnotations, Boolean suppressConnectionStringWarning) in C:\Users\Tony\Source\EntityFrameworkCore.Scaffolding.Handlebars\src\EntityFrameworkCore.Scaffolding.Handlebars\HbsCSharpDbContextGenerator.cs:line 127
```

**Comment by tonysneed** [comment]
@gejotas Are you able to make the changes I requested?

**Comment by gejotas** [comment]
> @gejotas Are you able to make the changes I requested?

@tonysneed I'll look at it either this weekend or next week.

**Comment by tonysneed** [comment]
@gejotas Since you created this PR, I have written contributing guidelines with step-by-step instructions for creating a PR. Before you make additional commits on your feature branch, I suggest you install Visual Studio Code and [configure Git](https://github.com/TrackableEntities/EntityFrameworkCore.Scaffolding.Handlebars/blob/master/.github/CONTRIBUTING.md#setup) to use it as its default editor. Then follow [Step 15](https://github.com/TrackableEntities/EntityFrameworkCore.Scaffolding.Handlebars/blob/master/.github/CONTRIBUTING.md#create-pull-request) to sync your fork with upstream commits.

Only _after I have reviewed and approved your PR_, follow [Step 16](https://github.com/TrackableEntities/EntityFrameworkCore.Scaffolding.Handlebars/blob/master/.github/CONTRIBUTING.md#squash-commits) to squash your commits.

**Comment by tonysneed** [comment]
@gejotas Any ETA?

**Comment by tonysneed** [comment]
Resolved issues and fixed bug in new PR #124.
